// react-redux-modal-flex/lib/modules/selectors.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isFooter = exports.disabledOk = exports.actionOk = exports.classOk = exports.textOk = exports.ok = exports.textCancel = exports.closeBtn = exports.title = exports.Component = exports.closeByMask = exports.mask = exports.duration = exports.animation = exports.classContent = exports.show = exports.modal = undefined;

var _reselect = require('reselect');

var modal = exports.modal = function modal(state) {
  return state.modal;
};

var show = exports.show = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.show;
});

var classContent = exports.classContent = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.classContent;
});

var animation = exports.animation = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.animation;
});

var duration = exports.duration = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.duration;
});

var mask = exports.mask = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.mask;
});

var closeByMask = exports.closeByMask = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.closeByMask;
});

var Component = exports.Component = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.component;
});

var title = exports.title = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.title;
});

var closeBtn = exports.closeBtn = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.closeBtn;
});

var textCancel = exports.textCancel = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.textCancel;
});

var ok = exports.ok = (0, _reselect.createSelector)(modal, function (modal) {
  return modal.ok;
});

var textOk = exports.textOk = (0, _reselect.createSelector)(ok, function (ok) {
  return ok.text;
});

var classOk = exports.classOk = (0, _reselect.createSelector)(ok, function (ok) {
  return ok.classOk;
});

var actionOk = exports.actionOk = (0, _reselect.createSelector)(ok, function (ok) {
  return ok.action;
});

var disabledOk = exports.disabledOk = (0, _reselect.createSelector)(ok, function (ok) {
  return ok.disabled;
});

var isFooter = exports.isFooter = (0, _reselect.createSelector)(textCancel, textOk, function (cancel, ok) {
  return !!(ok || cancel);
});
