// react-redux-modal-flex/lib/Modal.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _pick2 = require('ramda/src/pick');

var _pick3 = _interopRequireDefault(_pick2);

var _is2 = require('ramda/src/is');

var _is3 = _interopRequireDefault(_is2);

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _templateObject = _taggedTemplateLiteral(['\n  position: fixed;\n  width: 100%;\n  height: 100%;\n  top: 0px;\n  left: 0px;\n  overflow: hidden;\n  z-index: 9999;\n  display: ', ';\n'], ['\n  position: fixed;\n  width: 100%;\n  height: 100%;\n  top: 0px;\n  left: 0px;\n  overflow: hidden;\n  z-index: 9999;\n  display: ', ';\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _reactRedux = require('react-redux');

var _reselect = require('reselect');

var _styled = require('./styled');

var _modules = require('./modules');

var _Content = require('./components/Content');

var _Content2 = _interopRequireDefault(_Content);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents2.default.div(_templateObject, function (props) {
  return props.show ? 'block' : 'none';
});

var Modal = function (_React$Component) {
  _inherits(Modal, _React$Component);

  function Modal() {
    _classCallCheck(this, Modal);

    return _possibleConstructorReturn(this, (Modal.__proto__ || Object.getPrototypeOf(Modal)).apply(this, arguments));
  }

  _createClass(Modal, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          show = _props.show,
          classContent = _props.classContent,
          animation = _props.animation,
          Component = _props.Component,
          toggleModal = _props.toggleModal,
          modifyOkModal = _props.modifyOkModal;


      return _react2.default.createElement(
        _react2.default.Fragment,
        null,
        _react2.default.createElement(_styled.GlobalStyle, null),
        _react2.default.createElement(
          Wrapper,
          { show: show },
          _react2.default.createElement(
            _Content2.default,
            _extends({
              className: (0, _classnames2.default)('animated', classContent, animation)
            }, this.props),
            show && (0, _is3.default)(Function, Component) && _react2.default.createElement(Component, {
              toggleModal: toggleModal,
              modifyOkModal: modifyOkModal
            })
          )
        )
      );
    }
  }]);

  return Modal;
}(_react2.default.Component);

Modal.propTypes = {
  show: _propTypes2.default.bool.isRequired,
  Component: _propTypes2.default.func,
  classContent: _propTypes2.default.string.isRequired,
  animation: _propTypes2.default.string.isRequired,
  duration: _propTypes2.default.number.isRequired,
  mask: _propTypes2.default.bool.isRequired,
  closeByMask: _propTypes2.default.bool.isRequired,
  title: _propTypes2.default.string,
  closeBtn: _propTypes2.default.bool.isRequired,
  textOk: _propTypes2.default.string,
  classOk: _propTypes2.default.string.isRequired,
  actionOk: _propTypes2.default.func,
  disabledOk: _propTypes2.default.bool,
  textCancel: _propTypes2.default.string,
  isFooter: _propTypes2.default.bool.isRequired,
  toggleModal: _propTypes2.default.func.isRequired,
  modifyOkModal: _propTypes2.default.func.isRequired
};
exports.default = (0, _reactRedux.connect)((0, _reselect.createStructuredSelector)((0, _pick3.default)(['show', 'Component', 'classContent', 'animation', 'duration', 'mask', 'closeByMask', 'title', 'closeBtn', 'textOk', 'classOk', 'actionOk', 'disabledOk', 'textCancel', 'isFooter'], _modules.selectors)), (0, _pick3.default)(['toggleModal', 'modifyOkModal'], _modules.actions))(Modal);
