// react-redux-modal-flex/lib/components/Content/components/Footer/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  margin-top: auto;\n  padding: 10px;\n  border-top: 1px solid #e5e5e9;\n  display: flex;\n  justify-content: flex-end;\n  align-items: center;\n  flex: 1 0 auto;\n'], ['\n  margin-top: auto;\n  padding: 10px;\n  border-top: 1px solid #e5e5e9;\n  display: flex;\n  justify-content: flex-end;\n  align-items: center;\n  flex: 1 0 auto;\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Button = require('./components/Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents2.default.footer(_templateObject);

var Footer = function Footer(_ref) {
  var textOk = _ref.textOk,
      classOk = _ref.classOk,
      textCancel = _ref.textCancel,
      toggleModal = _ref.toggleModal,
      disabledOk = _ref.disabledOk,
      actionOk = _ref.actionOk;
  return _react2.default.createElement(
    Wrapper,
    null,
    textCancel && _react2.default.createElement(
      _Button2.default,
      { onClick: function onClick() {
          return toggleModal(false);
        } },
      textCancel
    ),
    textOk && _react2.default.createElement(
      _Button2.default,
      {
        primary: true,
        isDisabled: disabledOk,
        onClick: function onClick() {
          return actionOk && actionOk();
        },
        className: classOk
      },
      textOk
    )
  );
};

Footer.propTypes = {
  textOk: _propTypes2.default.string,
  classOk: _propTypes2.default.string.isRequired,
  textCancel: _propTypes2.default.string,
  toggleModal: _propTypes2.default.func.isRequired,
  actionOk: _propTypes2.default.func,
  disabledOk: _propTypes2.default.bool.isRequired,
  className: _propTypes2.default.string.isRequired
};

exports.default = Footer;
