// react-redux-modal-flex/lib/components/Content/components/Footer/components/Button/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  color: #83878d;\n  border-radius: 5px;\n  cursor: pointer;\n  padding: 8px 12px;\n  text-transform: uppercase;\n  transition: all 0.2s;\n  &:hover {\n    background: #f2fbff;\n    color: #6d7379;\n  }\n  &.primary {\n    color: #fc4a4a;\n  }\n  &.isDisabled {\n    pointer-events: none;\n    opacity: 0.6;\n  }\n'], ['\n  color: #83878d;\n  border-radius: 5px;\n  cursor: pointer;\n  padding: 8px 12px;\n  text-transform: uppercase;\n  transition: all 0.2s;\n  &:hover {\n    background: #f2fbff;\n    color: #6d7379;\n  }\n  &.primary {\n    color: #fc4a4a;\n  }\n  &.isDisabled {\n    pointer-events: none;\n    opacity: 0.6;\n  }\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents2.default.span(_templateObject);

var Button = function Button(_ref) {
  var children = _ref.children,
      isDisabled = _ref.isDisabled,
      onClick = _ref.onClick,
      primary = _ref.primary,
      className = _ref.className;
  return _react2.default.createElement(
    Wrapper,
    {
      className: (0, _classnames2.default)(className, { isDisabled: isDisabled, primary: primary }),
      onClick: onClick
    },
    children
  );
};

Button.propTypes = {
  children: _propTypes2.default.node.isRequired,
  primary: _propTypes2.default.bool,
  isDisabled: _propTypes2.default.bool,
  onClick: _propTypes2.default.func,
  className: _propTypes2.default.string
};

exports.default = Button;
