// react-redux-modal-flex/lib/components/Content/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _templateObject = _taggedTemplateLiteral(['\n  position: absolute;\n  width: 100%;\n  height: 100%;\n  top: 0px;\n  left: 0px;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  background: rgba(0, 0, 0, 0.5);\n  animation-duration: 0.2s;\n  & > div.animated {\n    display: flex;\n    flex-direction: column;\n    min-width: 320px;\n    max-width: 98vw;\n    max-height: 98vh;\n    background: #fff;\n    border-radius: 2px;\n    border: 1px solid #ebebee;\n    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.33);\n    animation-duration: ', 'ms;\n  }\n'], ['\n  position: absolute;\n  width: 100%;\n  height: 100%;\n  top: 0px;\n  left: 0px;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  background: rgba(0, 0, 0, 0.5);\n  animation-duration: 0.2s;\n  & > div.animated {\n    display: flex;\n    flex-direction: column;\n    min-width: 320px;\n    max-width: 98vw;\n    max-height: 98vh;\n    background: #fff;\n    border-radius: 2px;\n    border: 1px solid #ebebee;\n    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.33);\n    animation-duration: ', 'ms;\n  }\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Header = require('./components/Header');

var _Header2 = _interopRequireDefault(_Header);

var _Footer = require('./components/Footer');

var _Footer2 = _interopRequireDefault(_Footer);

var _Main = require('./components/Main');

var _Main2 = _interopRequireDefault(_Main);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents2.default.section(_templateObject, function (props) {
  return props.duration;
});

var Content = function (_React$Component) {
  _inherits(Content, _React$Component);

  function Content() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, Content);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Content.__proto__ || Object.getPrototypeOf(Content)).call.apply(_ref, [this].concat(args))), _this), _this.onClick = function (_ref2) {
      var target = _ref2.target;
      var _this$props = _this.props,
          mask = _this$props.mask,
          closeByMask = _this$props.closeByMask,
          toggleModal = _this$props.toggleModal;

      if (mask && closeByMask && /modal-overlay/i.test(target.className)) {
        toggleModal(false);
      }
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Content, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          mask = _props.mask,
          className = _props.className,
          children = _props.children,
          isFooter = _props.isFooter,
          duration = _props.duration;


      return _react2.default.createElement(
        Wrapper,
        {
          className: 'animated fadeIn modal-overlay',
          onClick: this.onClick,
          duration: duration
        },
        _react2.default.createElement(
          'div',
          { className: className },
          _react2.default.createElement(_Header2.default, this.props),
          _react2.default.createElement(
            _Main2.default,
            null,
            children
          ),
          isFooter && _react2.default.createElement(_Footer2.default, this.props)
        )
      );
    }
  }]);

  return Content;
}(_react2.default.Component);

Content.propTypes = {
  mask: _propTypes2.default.bool.isRequired,
  className: _propTypes2.default.string.isRequired,
  children: _propTypes2.default.node,
  isFooter: _propTypes2.default.bool.isRequired,
  duration: _propTypes2.default.number.isRequired,
  toggleModal: _propTypes2.default.func.isRequired
};
exports.default = Content;
